core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -973,6 +973,14 @@ private NativeImageSourceJarBuildItem buildNativeImageThinJar(CurateOutcomeBuild
 
             log.info("Building native image source jar: " + runnerJar);
 
+            // Remove svm and graal-sdk artifacts as they are provided by GraalVM itself
+            if (classLoadingConfig.removedArtifacts.isEmpty()) {
+                classLoadingConfig.removedArtifacts = Optional.of(new ArrayList<>(2));
+            }
+            List<String> removedArtifacts = classLoadingConfig.removedArtifacts.get();
+            removedArtifacts.add("org.graalvm.nativeimage:svm");
+            removedArtifacts.add("org.graalvm.sdk:graal-sdk");
```

Comment:
Can't say I'm excited about us actually updating the config. But I suppose it shouldn't be a huge problem in practice.

Reply:
I went this way to reuse the existing code and avoid introducing more parameters and checks that would essentially do the same thing. As this is not urgent please let me know if you can think of any better alternatives.